### PR TITLE
Default values for uitype

### PR DESCRIPTION
**Pull request by arek1-618**
Default values for uitype.

Improving translations:
firefox_2018-09-27_16-12-53

Improving the modal window:
before: 

after:

### CODE REVIEW

File: modules/Vtiger/uitypes/Date.php [hunk 1]
```diff
@@ -127,4 +127,17 @@ public function getAllowedColumnTypes()
...
+	 * {@inheritdoc}
+	 */
+	public function setDefaultValueFromRequest(\App\Request $request)
+	{
+		$fieldName = $this->getFieldModel()->getFieldName();
+		$value = $request->get($fieldName);
+		if (!\App\TextParser::isVaribleToParse($value)) {
+			$this->validate($value, true);
+		}
+		$this->getFieldModel()->set('defaultvalue', $this->getDBValue($value));
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Date.php]: ??

File: modules/Vtiger/uitypes/Date.php [hunk 2]
```diff
@@ -127,4 +127,17 @@ public function getAllowedColumnTypes()
...
+	 * {@inheritdoc}
+	 */
+	public function setDefaultValueFromRequest(\App\Request $request)
+	{
+		$fieldName = $this->getFieldModel()->getFieldName();
+		$value = $request->getByType($fieldName, 'Text');
+		if (!\App\TextParser::isVaribleToParse($value)) {
+			$this->validate($value, true);
+		}
+		$this->getFieldModel()->set('defaultvalue', $value);
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Date.php]: Add $this->getDBValue($value)

File: tests/App/FieldModel.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	/**
+	 * Test validate for wrong data.
+	 *
+	 * @param int    $uiType
+	 * @param string $typeOfData
+	 * @param string $columnType
+	 * @param mixed  $value
+	 * @dataProvider providerWrongData
+	 */
+	public function testValidateWrongeData(int $uiType, string $typeOfData, string $columnType, $value)
```
Comment (slawekk87) [inline tests/App/FieldModel.php]: Fix typo Wronge => Wrong

File: tests/App/FieldModel.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+	/**
+	 * Test validate for user format.
+	 *
+	 * @param int    $uiType
+	 * @param string $typeOfData
+	 * @param string $columnType
+	 * @param mixed  $value
+	 * @dataProvider providerWrongData
+	 */
+	public function testValidateUserFormatWrongeData(int $uiType, string $typeOfData, string $columnType, $value)
```
Comment (slawekk87) [inline tests/App/FieldModel.php]: Fix typo Wronge => Wrong

File: tests/App/FieldModel.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+	 */
+	protected static $fields;
+
+	/**
+	 * Create Vtiger_Field_Model for tests.
+	 *
+	 * @param int    $uiType
+	 * @param string $typeOfData
+	 * @param string $columnType
+	 * @param mixed  $value
```
Comment (slawekk87) [inline tests/App/FieldModel.php]: ??

File: layouts/basic/modules/Settings/LayoutEditor/VaribleToParsers.tpl [hunk 6]
```diff
@@ -1,17 +1,18 @@
 {*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
 {strip}
-	<div class="modal-header">
-		<h5 class="modal-title">{App\Language::translate('LBL_CUSTOM_VARIABLES', $QUALIFIED_MODULE)}</h5>
+	<div class="tpl-Settings-LayoutEditor-VaribleToParsers modal-header">
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Settings/LayoutEditor/VaribleToParsers.tpl]: `<!-- tpl-Settings-LayoutEditor-VaribleToParsers -->`

**Pull request merged**